support-models/src/main/scala/com/gu/support/workers/PaymentFields.scala

```diff
@@ -45,6 +67,15 @@ object PaymentFields {
       Decoder[StripePaymentMethodPaymentFields].widen,
       Decoder[DirectDebitPaymentFields].widen,
       Decoder[ExistingPaymentFields].widen
-    ).reduceLeft(_ or _)
+    ).reduceLeft(or(_,_))
+
+  final def or[A, AA >: A](a: Decoder[A], d: => Decoder[AA]): Decoder[AA] = new Decoder[AA] {
+    final def apply(c: HCursor): Decoder.Result[AA] = a(c) match {
+      case r @ Right(_) => r
+      case Left(err)      => d(c).left.map { (decodingFailure: DecodingFailure) =>
+        DecodingFailure(err.message + " OR " + decodingFailure.message, err.history ++ decodingFailure.history)
```

Comment:
this is not great, all the information is there if you squint, but it's all mashed together a bit.  It got me where I needed to be though.